Gemfile

```diff
@@ -2,3 +2,11 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in kafo_parsers.gemspec
 gemspec
+
+puppet_version = ENV['PUPPET_VERSION'] || ENV['puppet']
+puppet_spec = puppet_version ? "~> #{puppet_version}" : '< 5.0.0'
+gem 'puppet', puppet_spec
+
+if puppet_version.nil? || puppet_version >= '3.7'
```

Comment:
Yeah, should do for any typical string that's likely to be entered here.  I can't think of an example that wouldn't.